impl/events.go

```diff
@@ -170,12 +168,42 @@ func (m *manager) OnRequestTimedOut(ctx context.Context, chid datatransfer.Chann
 	go func() {
 		select {
 		case <-ctx.Done():
-		case <-time.After(ChannelRemoveTimeout):
+		case <-time.After(m.channelRemoveTimeout):
+			channel, err := m.channels.GetByID(ctx, chid)
```

Comment:
What happens if a channel times/out disconnects and then the FSM/data-trasfer crashes ? I think we should have a handler for this state that starts this wait even when the FSM restarts. Does that make sense ?

Reply:
yea we can address this when we have a better way to actually track disconencts other than the Message -- see comment above about challenges of essentially pushing state